### PR TITLE
create_and_store/2, update_and_store/2,

### PR DESCRIPTION
**Pull request by tashirosota**

### CODE REVIEW

File: lib/ecto_cellar.ex [hunk 1]
```diff
@@ -38,28 +39,90 @@ defmodule EctoCellar do
...
+    |> Multi.run(:store, fn _repo, %{model: model} -> store(model, opts) end)
+    |> repo(opts).transaction()
+    |> case do
+      {:ok, %{model: model}} ->
+        {:ok, model}
+
+      error ->
+        error
+    end
+  end
```
Comment (tashirosota) [inline lib/ecto_cellar.ex:76]: @mokichi
I tried wrapping some functions of Ecto.Repo.
 - insert_and_store
 - update_and_store
 - insert_or_update_and_store

I think tracking will be easier if there is something around here, but what about?

**Comment by tashirosota** [comment]
https://github.com/tashirosota/ecto_cellar/issues/13

File: lib/ecto_cellar.ex (hunk 1 above)
Comment (mokichi) [reply lib/ecto_cellar.ex]: @tashirosota Great! But I think `delete_and_store` is also needed. In addition, `upsert_store` is more concise than `insert_or_update_and_store`.

Comment (tashirosota) [reply lib/ecto_cellar.ex]: @mokichi 
Thanks for response！
`upsert_store` is too nice！ I'll rename it soon！

> delete_and_store is also needed.

The stored mechanism of Ecto is like this.

 1. creates and stores to versions table.
 2. updates and stores to versions table.

In each step, the state is stored after issuing the insert of the model.
In other words, if the model itself does not exist with delete, it is difficult to store them.

Do you have any good ideas?

Comment (tashirosota) [reply lib/ecto_cellar.ex]: Maybe I have to prepare a table to record the deletions.

Comment (tashirosota) [reply lib/ecto_cellar.ex]: nnn

Comment (tashirosota) [reply lib/ecto_cellar.ex]: Maybe it's enough to have this?
```elixir
  def delete_store(changeset, opts \\ [], delete_opts \\ []),
    do: do_wrap_func(changeset, opts, delete_opts, :delete)

  defp do_wrap_func(changeset, celler_opts, ecto_opts, func_atom) do
    Multi.new()
    |> Multi.run(:schema, fn _repo, _ ->
      apply(repo(celler_opts), func_atom, [changeset, ecto_opts])
    end)
    |> Multi.run(:store, fn _repo, %{schema: schema} -> store(schema, celler_opts) end)
    |> repo(celler_opts).transaction()
    |> case do
      {:ok, %{schema: schema}} ->
        {:ok, schema}

      error ->
        error
    end
  end
```

Comment (tashirosota) [reply lib/ecto_cellar.ex]: https://github.com/tashirosota/ecto_cellar/pull/14/files

**Pull request merged**